internal/config/cgroupmanager/cgroupfsmanager.go

```diff
@@ -0,0 +1,84 @@
+// +build linux
+
+package cgroupmanager
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/libpod/pkg/cgroups"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+type CgroupfsManager struct{}
+
+const (
+	defaultCgroupfsParent = "/crio"
+)
+
+// Name returns the name of the cgroup manager (cgroupfs)
+func (*CgroupfsManager) Name() string {
+	return CgroupfsCgroupManager
+}
+
+// IsSystemd returns that this is not a systemd cgroup manager
+func (*CgroupfsManager) IsSystemd() bool {
+	return false
+}
+
+// GetContainerCgroupPath takes arguments sandbox parent cgroup and container ID and returns
+// the cgroup path for that containerID. If parentCgroup is empty, it
+// uses the default parent /crio
+func (*CgroupfsManager) GetContainerCgroupPath(sbParent, containerID string) string {
+	parent := defaultCgroupfsParent
+	if sbParent != "" {
+		parent = sbParent
+	}
+	return filepath.Join(parent, scopePrefix+"-"+containerID)
+}
+
+// GetSandboxCgroupPath takes the sandbox parent, and sandbox ID. It
+// returns the cgroup parent, cgroup path, and error.
+func (*CgroupfsManager) GetSandboxCgroupPath(sbParent, sbID string) (string, string, error) {
+	if strings.HasSuffix(path.Base(sbParent), ".slice") {
+		return "", "", fmt.Errorf("cri-o configured with cgroupfs cgroup manager, but received systemd slice as parent: %s", sbParent)
+	}
+	return sbParent, filepath.Join(sbParent, scopePrefix+"-"+sbID), nil
+}
+
+// MoveConmonToCgroup takes the container ID, cgroup parent, conmon's cgroup (from the config) and conmon's PID
+// It attempts to move conmon to the correct cgroup.
+// It returns the cgroupfs parent that conmon was put into
+// so that CRI-O can clean the cgroup path of the newly added conmon once the process terminates (systemd handles this for us)
+func (*CgroupfsManager) MoveConmonToCgroup(cid, cgroupParent, conmonCgroup string, pid int) string {
+	// Move conmon to specified cgroup
+	if conmonCgroup != "pod" && conmonCgroup != "" {
+		logrus.Errorf("conmon cgroup %s invalid for cgroupfs", conmonCgroup)
+		return ""
+	}
+
+	cgroupPath := filepath.Join(cgroupParent, "/crio-conmon-"+cid)
```

Comment:
having both `/` and `filepath.Join` is redundant -- you can drop either one. I would drop `filepath.Join` but either way is fine.